## Number of PDR Requests Per Year (2014-2018)
#### Question: 
Have the Number of Requests Increased Over the Last Four Years? 
#### Plan:
Count Frequency of Years as Recored by RID Field in Dataset and Plot 
#### Caviat: 
Year 2018 unfinshed: Dataset is based on requests as of 6/17/2018 will take into account in conclusion 

### Import Data 

In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv("/Users/curcian/Documents/Week 5/PDR-csv.csv")
df

Column            Assigned    AssignedGroup CollisionDenialYN  \
0           0       Matt Clements       FMS_PDR_OR               NaN   
1           1     Elena Bakhareva              NaN               NaN   
2           2        Joni Higgins      FMS_PDR_HQR               NaN   
3           3      Patricia Smith  FMS_PDR_TOLLING               NaN   
4           4        Joni Higgins      FMS_PDR_HQR               NaN   
5           5        Kate Duckett      FMS_PDR_HQR               NaN   
6           6     Eric Van Hooser      FMS_PDR_NWR               NaN   
7           7     Tanya Joblonski      FMS_PDR_TDO               NaN   
8           8        Kate Duckett      FMS_PDR_HQR               NaN   
9           9       Matt Clements       FMS_PDR_OR               NaN   
10         10       Matt Clements       FMS_PDR_OR               NaN   
11         11   Celeste Dimichina      FMS_PDR_SWR               NaN   
12         12     Eric Van Hooser      FMS_PDR_NWR               NaN   
13         13           Al Gilson       FMS_PDR_ER               NaN   
14         14     Eric Van Hooser      FMS_PDR_NWR               Yes   
15         15           Al Gilson       FMS_PDR_ER               NaN   
16         16  April BickarPierce      FMS_PDR_HQR               NaN   
17         17        Kate Duckett      FMS_PDR_HQR               NaN   
18         18        Joni Higgins      FMS_PDR_HQR               NaN   
19         19        Kate Duckett      FMS_PDR_HQR               NaN   
20         20     Eric Van Hooser      FMS_PDR_NWR               NaN   
21         21     Eric Van Hooser      FMS_PDR_NWR               NaN   
22         22   Celeste Dimichina      FMS_PDR_SWR               NaN   
23         23     Eric Van Hooser      FMS_PDR_NWR               NaN   
24         24     Eric Van Hooser      FMS_PDR_NWR               NaN   
25         25   Celeste Dimichina      FMS_PDR_SWR               NaN   
26         26          Fred Chang      FMS_PDR_UCO               NaN   
27         27     Eric Van Hooser      FMS_PDR_NWR               NaN   
28         28     Patricia Thomas      FMS_PDR_SCR               NaN   
29         29     Eric Van Hooser      FMS_PDR_NWR               NaN   
...       ...                 ...              ...               ...   
15635   15639           Bryan Ray  FMS_PDR_HQR_ENR               NaN   
15636   15640          Fred Chang      FMS_PDR_UCO               NaN   
15637   15641        Melinda Ross      FMS_PDR_HQR               NaN   
15638   15642                 NaN      FMS_PDR_HQR               NaN   
15639   15643          Robert Lee      FMS_PDR_HQR               NaN   
15640   15644                 NaN      FMS_PDR_HQR               NaN   
15641   15645      Patricia Smith  FMS_PDR_TOLLING               NaN   
15642   15646                 NaN      FMS_PDR_HQR               NaN   
15643   15647                 NaN      FMS_PDR_HQR               NaN   
15644   15648   Allison Bergstrom      FMS_PDR_NCR               NaN   
15645   15649                 NaN      FMS_PDR_HQR               NaN   
15646   15650                 NaN      FMS_PDR_HQR               NaN   
15647   15651                 NaN      FMS_PDR_HQR               NaN   
15648   15652                 NaN      FMS_PDR_HQR               NaN   
15649   15654                 NaN      FMS_PDR_HQR               NaN   
15650   15655          Robert Lee      FMS_PDR_HQR               NaN   
15651   15656          Robert Lee      FMS_PDR_HQR               NaN   
15652   15657                 NaN      FMS_PDR_HQR               NaN   
15653   15658                 NaN      FMS_PDR_HQR               NaN   
15654   15659           Bryan Ray  FMS_PDR_HQR_ENR               NaN   
15655   15660          Robert Lee      FMS_PDR_HQR               NaN   
15656   15661      Patricia Smith  FMS_PDR_TOLLING               NaN   
15657   15662                 NaN      FMS_PDR_HQR               NaN   
15658   15663      Patricia Smith  FMS_PDR_TOLL

### Data Cleaning
Two different ways RID has been standardized 
PDR-15-2589
 -161285
Will convert to all digits  

In [2]:
df["RID"].astype(str)

0        PDR-15-2589
1            -161285
2        PDR-15-2343
3            -173047
4        PDR-13-1990
5            -171723
6            -162929
7            -163239
8            -161730
9        PDR-14-3449
10       PDR-15-2653
11           -171600
12           -161825
13           -180301
14       PDR-14-2305
15           -180930
16           -163613
17           -181485
18       PDR-15-0560
19           -172019
20       PDR-14-1016
21           -163079
22       PDR-15-3329
23           -172225
24       PDR-15-2477
25           -160172
26       PDR-14-3224
27           -162308
28           -180310
29       PDR-14-1214
            ...     
15635        -172628
15636        -172695
15637        -172820
15638        -172864
15639        -173034
15640        -173046
15641        -173053
15642        -173129
15643        -173145
15644        -173288
15645        -173366
15646        -180077
15647        -180249
15648        -180327
15649        -180428
15650        -180554
15651        

In [3]:
df["RID"].replace(regex=True, inplace=True, to_replace=r'\D',value=r"")

In [4]:
df["RID"]

0        152589
1        161285
2        152343
3        173047
4        131990
5        171723
6        162929
7        163239
8        161730
9        143449
10       152653
11       171600
12       161825
13       180301
14       142305
15       180930
16       163613
17       181485
18       150560
19       172019
20       141016
21       163079
22       153329
23       172225
24       152477
25       160172
26       143224
27       162308
28       180310
29       141214
          ...  
15635    172628
15636    172695
15637    172820
15638    172864
15639    173034
15640    173046
15641    173053
15642    173129
15643    173145
15644    173288
15645    173366
15646    180077
15647    180249
15648    180327
15649    180428
15650    180554
15651    180694
15652    180711
15653    180794
15654    180800
15655    180830
15656    180913
15657    181164
15658    181258
15659    181277
15660    181287
15661    181354
15662    181394
15663    181415
15664    181611
Name: RID, Length: 15665

#### Analysis 
Create New Year Column Based on Clean RIDs and Find Frequency of Year 

In [5]:
df.reset_index(inplace=True)

In [6]:
def year(RID):
    for i in RID:
        if RID.startswith("14"):
            return "2014"
        elif RID.startswith("15"):
            return "2015"
        elif RID.startswith("16"):
            return "2016"
        elif RID.startswith("17"):
            return "2017"
        elif RID.startswith("18"):
            return "2018"
        else: 
            return "NaN"
        
df["Year"] = df.RID.apply(year)
df
    

index  Column            Assigned    AssignedGroup CollisionDenialYN  \
0          0       0       Matt Clements       FMS_PDR_OR               NaN   
1          1       1     Elena Bakhareva              NaN               NaN   
2          2       2        Joni Higgins      FMS_PDR_HQR               NaN   
3          3       3      Patricia Smith  FMS_PDR_TOLLING               NaN   
4          4       4        Joni Higgins      FMS_PDR_HQR               NaN   
5          5       5        Kate Duckett      FMS_PDR_HQR               NaN   
6          6       6     Eric Van Hooser      FMS_PDR_NWR               NaN   
7          7       7     Tanya Joblonski      FMS_PDR_TDO               NaN   
8          8       8        Kate Duckett      FMS_PDR_HQR               NaN   
9          9       9       Matt Clements       FMS_PDR_OR               NaN   
10        10      10       Matt Clements       FMS_PDR_OR               NaN   
11        11      11   Celeste Dimichina      FMS_PDR_SWR               NaN   
12        12      12     Eric Van Hooser      FMS_PDR_NWR               NaN   
13        13      13           Al Gilson       FMS_PDR_ER               NaN   
14        14      14     Eric Van Hooser      FMS_PDR_NWR               Yes   
15        15      15           Al Gilson       FMS_PDR_ER               NaN   
16        16      16  April BickarPierce      FMS_PDR_HQR               NaN   
17        17      17        Kate Duckett      FMS_PDR_HQR               NaN   
18        18      18        Joni Higgins      FMS_PDR_HQR               NaN   
19        19      19        Kate Duckett      FMS_PDR_HQR               NaN   
20        20      20     Eric Van Hooser      FMS_PDR_NWR               NaN   
21        21      21     Eric Van Hooser      FMS_PDR_NWR               NaN   
22        22      22   Celeste Dimichina      FMS_PDR_SWR               NaN   
23        23      23     Eric Van Hooser      FMS_PDR_NWR               NaN   
24        24      24     Eric Van Hooser      FMS_PDR_NWR               NaN   
25        25      25   Celeste Dimichina      FMS_PDR_SWR               NaN   
26        26      26          Fred Chang      FMS_PDR_UCO               NaN   
27        27      27     Eric Van Hooser      FMS_PDR_NWR               NaN   
28        28      28     Patricia Thomas      FMS_PDR_SCR               NaN   
29        29      29     Eric Van Hooser      FMS_PDR_NWR               NaN   
...      ...     ...                 ...              ...               ...   
15635  15635   15639           Bryan Ray  FMS_PDR_HQR_ENR               NaN   
15636  15636   15640          Fred Chang      FMS_PDR_UCO               NaN   
15637  15637   15641        Melinda Ross      FMS_PDR_HQR               NaN   
15638  15638   15642                 NaN      FMS_PDR_HQR               NaN   
15639  15639   15643          Robert Lee      FMS_PDR_HQR               NaN   
15640  15640   15644                 NaN      FMS_PDR_HQR               NaN   
15641  15641   15645      Patricia Smith  FMS_PDR_TOLLING               NaN   
15642  15642   15646                 NaN      FMS_PDR_HQR               NaN   
15643  15643   15647                 NaN      FMS_PDR_HQR               NaN   
15644  15644   15648   Allison Bergstrom      FMS_PDR_NCR               NaN   
15645  15645   15649                 NaN      FMS_PDR_HQR               NaN   
15646  15646   15650                 NaN      FMS_PDR_HQR               NaN   
15647  15647   15651                 NaN      FMS_PDR_HQR               NaN   
15648  15648   15652                 NaN      FMS_PDR_HQR               NaN   
15649  15649   15654                 NaN      FMS_PDR_HQR               NaN   
15650  15650   15655          Robert Lee      FMS_PDR_HQR               NaN   
15651  15651   15656          Robert Lee      FMS_PDR_HQR               NaN   
15652  15652   15657                 NaN      FMS_PDR_HQR               NaN   
15653  15653   15658                 NaN      FMS_PDR_HQR

In [7]:
df["RID"].astype(str)

0        152589
1        161285
2        152343
3        173047
4        131990
5        171723
6        162929
7        163239
8        161730
9        143449
10       152653
11       171600
12       161825
13       180301
14       142305
15       180930
16       163613
17       181485
18       150560
19       172019
20       141016
21       163079
22       153329
23       172225
24       152477
25       160172
26       143224
27       162308
28       180310
29       141214
          ...  
15635    172628
15636    172695
15637    172820
15638    172864
15639    173034
15640    173046
15641    173053
15642    173129
15643    173145
15644    173288
15645    173366
15646    180077
15647    180249
15648    180327
15649    180428
15650    180554
15651    180694
15652    180711
15653    180794
15654    180800
15655    180830
15656    180913
15657    181164
15658    181258
15659    181277
15660    181287
15661    181354
15662    181394
15663    181415
15664    181611
Name: RID, Length: 15665

In [8]:
df["Year"].value_counts()

2016    3583
2015    3480
2014    3412
2017    3371
2018    1745
NaN       74
Name: Year, dtype: int64

In [9]:
bar = df["Year"].value_counts().plot('bar')

In [10]:
1745 * 2

3490

### Results 
No marked increase in PRD requests over last four years accoding to this analysis

Year |  Request Recorded
-----|----------------
2014 | 3412
2015 | 3480
2016 | 3583
2017 | 3371
2018* | 3490
*(Projected)|

##### Next Steps
Are Types of Requests and Requestors Changing? 